### PR TITLE
fix: Handle multiple messages coming all at once

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/plist-service/index.js [hunk 1]
```diff
@@ -28,10 +33,23 @@ class PlistService {
...
-    this._splitter.resume();
     return await new B((resolve, reject) => {
-      setTimeout(function () { reject(new Error(`Failed to receive any data within the timeout: ${timeout}`)); }, timeout);
-      this._decoder.once('data', (data) => resolve(data));
+      const queue = this.replyQueue;
+      const data = queue.shift();
+      if (data) {
+        resolve(data);
+      }
+      const checkExist = setInterval(() => {
```
Comment (imurchie) [inline lib/plist-service/index.js]: Should this be happening even if there was already data?

File: lib/usbmux/index.js [hunk 2]
```diff
@@ -84,22 +88,23 @@ class Usbmux {
     this._encoder.write(json);
   }
 
-  receivePlist (timeout = 5000) {
-    this._splitter.resume();
+  receivePlistPromise (timeout = 5000) {
     return new B((resolve, reject) => {
-      setTimeout(function () { reject(new Error(`Failed to receive any data within the timeout: ${timeout}`)); }, timeout);
       this._decoder.once('data', (data) => resolve(data));
```
Comment (imurchie) [inline lib/usbmux/index.js]: `this._decoder.once('data', resolve);`

File: lib/plist-service/index.js (hunk 1 above)
Comment (umutuzgur) [reply lib/plist-service/index.js]: Good catch. I should put a return before

File: lib/usbmux/index.js (hunk 2 above)
Comment (umutuzgur) [reply lib/usbmux/index.js]: Done

File: lib/plist-service/index.js [hunk 3]
```diff
@@ -28,10 +33,25 @@ class PlistService {
...
 
   async receivePlist (timeout = 5000) {
-    this._splitter.resume();
     return await new B((resolve, reject) => {
-      setTimeout(function () { reject(new Error(`Failed to receive any data within the timeout: ${timeout}`)); }, timeout);
-      this._decoder.once('data', (data) => resolve(data));
+      const queue = this.replyQueue;
+      const data = queue.shift();
+      if (data) {
+        resolve(data);
```
Comment (mykola-mokhnach) [inline lib/plist-service/index.js]: `return resolve(data)` would be shorter

**Pull request merged**